uberfire-services/uberfire-services-backend/src/main/java/org/guvnor/common/services/backend/config/AppConfigServiceImpl.java

```diff
@@ -54,28 +60,51 @@ public void configureOnEvent(@Observes ApplicationStarted applicationStartedEven
     public synchronized Map<String, String> loadPreferences() {
         try {
             if (preferences == null) {
-                preferences = new HashMap<String, String>();
-
-                //Load preferences from all stores
-                if (preferencesLoaders != null) {
-                    for (ApplicationPreferencesLoader loader : preferencesLoaders) {
-                        preferences.putAll(loader.load());
-                    }
-                }
-
-                //Perform any post-load handling of preferences
-                if (systemPropertiesInitializers != null) {
-                    for (SystemPropertiesInitializer initializer : systemPropertiesInitializers) {
-                        initializer.setSystemProperties(preferences);
-                    }
-                }
+                preferences = new HashMap<>();
+
+                // Load preferences from all stores
+                loadPreferencesFromAllStores();
+
+                // Load system properties (only to override previously loaded preferences, if necessary)
+                loadSystemProperties();
+
+                // Perform any post-load handling of preferences
+                setupSystemPropertiesInitializers();
             }
             return preferences;
         } catch (Exception e) {
             throw ExceptionUtilities.handleException(e);
         }
     }
 
+    private void loadPreferencesFromAllStores() {
+        if (preferencesLoaders != null) {
+            for (ApplicationPreferencesLoader loader : preferencesLoaders) {
+                preferences.putAll(loader.load());
+            }
+        }
+    }
+
+    private void loadSystemProperties() {
+        for (Map.Entry<String, String> entry : preferences.entrySet()) {
+            final String key = entry.getKey();
+            final String value = entry.getValue();
+            final String newValue = System.getProperty(key, value);
+
+            if (value != null && !value.equals(newValue)) {
+                preferences.put(key, newValue);
```

Comment:
Hello @Paulovmr,
The code looks and works fine. I only have one small objection about logging consistency. 

The DefaultApplicationPreferencesLoader logs the following on container startup:

```
09:17:33,311 INFO  [org.kie.workbench.common.services.backend.preferences.KieMetaDataPreferencesLoader] (MSC service thread 1-4) Setting preference 'kie_productized' to 'false'.
09:17:33,313 INFO  [org.kie.workbench.common.screens.examples.backend.server.ExamplesPreferencesLoader] (MSC service thread 1-4) Setting preference 'org.kie.demo' to 'true'.
09:17:33,344 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'designer.context' to 'designer'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'designer.profile' to 'jbpm'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'drools.defaultlanguage' to 'en'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'build.enable-incremental' to 'true'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'support.runtime.deploy' to 'true'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'drools.dateformat' to 'dd-MMM-yyyy'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'drools.datetimeformat' to 'dd-MMM-yyyy HH:mm:ss'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'rule-modeller-onlyShowDSLStatements' to 'false'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'drools.defaultcountry' to 'US'.
```

Don't you think we should add something similar to line 95 here to make it visible that some of the properties are being overriden?
Something like
"Overriding preference 'X' based on value from system property 'Y'."

Reply:
@jhrcek I agree, done. Thanks!